### DISCUSSION TITLE
How to create this kind of (Violin?) Graph?

### DISCUSSION THREAD

**Discussion by pyth0on** [theOehrly/Fast-F1#224] [Q&A]
Hi, i am currently learning how to use fastf1 and some people are creating graphs looking like this, but when I try to plot a violin graph I juts get error messages. What data do I have to request for these kind of graphs?
![Screenshot 2022-08-16 212051](https://user-images.githubusercontent.com/65238629/184964421-9797233f-c60a-47d6-b918-cada4e8d0bb7.png)

**Comment by theOehrly** [comment]
I'd say you only need lap times to generate this graph. But I can't really help you with how you create this graph. This is just a question of statistics and visualization, likely not related to FastF1 directly. I've never tried to create a visualization like this.

**Comment by TracingInsights** [comment]
This looks like it is created using matplotlib, but it is easier to visualize violin plots using seaborn library.

Checkout: https://seaborn.pydata.org/generated/seaborn.violinplot.html#seaborn.violinplot

You only need lap times of each team, but there is a caveat to this analysis. Since most teams don't always run the same strategy or tyres, it is important to account for fuel for a better understanding. I tried something similar here: https://www.reddit.com/r/TracingInsights/comments/wqx0kb/hungariangp_distribution_of_fuel_corrected_lap/
